### PR TITLE
Mergin api cleanups 4

### PR DESCRIPTION
**Pull request by wonder-sk**
This should conclude the refactoring...

Description in the commit message of 991254d

### CODE REVIEW

File: app/projectsmodel.cpp [hunk 1]
```diff
@@ -33,20 +33,49 @@ ProjectModel::~ProjectModel() {}
...
+      projectFile.info = "invalid project";
+      projectFile.isValid = false;
+    }
+    mProjectFiles << projectFile;
+  }
+
   std::sort( mProjectFiles.begin(), mProjectFiles.end() );
 }
 
+#if 0
```
Comment (sklencar) [inline app/projectsmodel.cpp]: Delete or keep if needed

File: app/projectsmodel.cpp [hunk 2]
```diff
@@ -220,7 +243,8 @@ void ProjectModel::addProject( QString projectFolder, QString projectName, bool
 
   Q_UNUSED( projectName );
   beginResetModel();
-  addProjectFromPath( projectFolder );
-  std::sort( mProjectFiles.begin(), mProjectFiles.end() );
+  //addProjectFromPath( projectFolder );
```
Comment (sklencar) [inline app/projectsmodel.cpp]: Commented out code. Same the line below

File: app/projectsmodel.cpp [hunk 3]
```diff
@@ -33,20 +33,49 @@ ProjectModel::~ProjectModel() {}
 
 void ProjectModel::findProjectFiles()
 {
+  /*
```
Comment (sklencar) [inline app/projectsmodel.cpp]: Please remove, if not needed

File: app/projectsmodel.cpp [hunk 4]
```diff
@@ -33,20 +33,49 @@ ProjectModel::~ProjectModel() {}
...
+  // populate from mLocalProjects
+  mProjectFiles.clear();
+  const QList<LocalProjectInfo> projects = mLocalProjects.projects();
+  for ( const LocalProjectInfo &project : projects )
+  {
+    QDir dir( project.projectDir );
+    QFileInfo fi( project.qgisProjectFilePath );
+
+    ProjectFile projectFile;
+    projectFile.name = fi.fileName().remove( ".qgs" ).remove( ".qgz" );   // TODO: shouldn't this be mergin project name??
```
Comment (sklencar) [inline app/projectsmodel.cpp]: Same as in Mergin, a project folder name is a name of the project. A real project filename isn't used.

File: app/merginapi.cpp [hunk 5]
```diff
@@ -1875,7 +1837,7 @@ MerginProject MerginApi::readProjectMetadata( const QByteArray &data )
     project.serverUpdated = QDateTime::fromString( docObj.value( QStringLiteral( "updated" ) ).toString(), Qt::ISODateWithMs ).toUTC();
     if ( project.version.isEmpty() )
     {
-      project.version = QStringLiteral( "v1" );
+      project.version = QStringLiteral( "v0" );
```
[the remark this answers is not in the document]
Comment (wonder-sk) [reply app/merginapi.cpp]: this is a fix for upload where the versions would go 2, 3, 4, ...

**Pull request merged**